src/internal/operations/inject/inject.go

```diff
@@ -59,6 +60,25 @@ type (
 		CacheItemInfo(v details.ItemInfo)
 	}
 
+	PopulateProtectedResourceIDAndNamer interface {
+		// PopulateProtectedResourceIDAndName takes the provided owner identifier and produces
+		// the owner's name and ID from that value.  Returns an error if the owner is
+		// not recognized by the current tenant.
+		//
+		// The id-name swapper should be optional.  Some processes will look up all owners in
```

Comment:
nit: Did you mean to say `idname.Cacher should be optional`?

Reply:
fixed, thank you